### PR TITLE
Milestone 5

### PR DESCRIPTION
**Pull request by sergiocortessat**
Pull request 5

Hello TSE. In this milestone, I have implemented the last required steps for the Calculator, such as adding the handle click events to App.js, Button.js, and panelButton.js. As such, they are in charge, taking care to listen each button's click and perform the correspondent activity. As well, I have added a simple extra feature with the % button. As such, it can convert the first value into a percentage, or, given the case, it will return the operation from 2 numbers and then convert it to percentage. Also, I decided to add some personal style and CSS to it.

- [x] App.js handle click
- [x] button.js click handler
- [x] buttonPanel.js click handler
- [x] Style component
- [x] Logic for a complete calculator

### CODE REVIEW

File: src/Components/App.js [hunk 1]
```diff
@@ -45,13 +45,16 @@ export default class App extends Component {
       result = next === null ? operation : next;
     }
```
Comment (mimipeshy) [inline src/Components/App.js]: - [x] Your code should be DRY. The whole section should be implemented in `logic.js`. Kindly work on that

File: src/Components/App.js [hunk 2]
```diff
@@ -45,13 +45,16 @@ export default class App extends Component {
...
-        <div>
-          <Display result={result} />
-          <ButtonPanel handleClick={this.handleClick} />
+      <>
+        <h2 className="title">Welcome to math magicians</h2>
+        <div className="app">
+          <h4 className="title-2">The Matrix Calculator</h4>
+          <div>
+            <Display result={result} />
+            <ButtonPanel handleClick={this.handleClick} />
```
Comment (mimipeshy) [inline src/Components/App.js]: - [x] From the instructions in pathwright, the `total` or `next` value should be passed from state to `Display` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone5/app/passToatalToDisplay.png))
- [x]  From the instructions in pathwright `handleClick` is passed as a `clickHandler` property to `ButtonPanel` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone5/app/passHandleClick.png))

File: src/Components/ButtonPanel.js [hunk 3]
```diff
@@ -11,7 +11,7 @@ const buttonGroups = {
 };
 
 const ButtonPanel = (props) => (
```
Comment (mimipeshy) [inline src/Components/ButtonPanel.js:13]: - [x] From the instructions from pathwright, `handleClick` is implemented as a function ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone5/panel/implementHandleClick.png))
- [x] Then  `handleClick` is passed as `clickHandler` props to each `Button` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone5/panel/passToButton.png))

**Review by mimipeshy — CHANGES_REQUESTED** [review]
## STATUS : CHANGES REQUESTED 🔴 ♻️ 
Hello @sergiocortessat and @ImgBotApp 👋🏾 👋🏾 congratulations on making it this far 😄 . 

#### Highlights

-  Extra credits for styling your app 👏🏾 ✔️ 
- Your live version works well ✔️ 

#### To improve
Even though your app works well, a few instructions were not implemented as expected ⬇️ 
_Check the comments under the review._

That's all 😄, kindly submit another review when done
If you need any clarification, kindly post your comment here and tag me @mimipeshy and will be glad to help

Regards,
Peris Ndanu

**Comment by sergiocortessat** [comment]
Pull request 6

Hello TSE. According to the changes requested, I managed to implement some of them as the remaining one will cause my app to be inefficient and it will require a unnecessary and memory loading to the page. I have implemented most of them. 

- Your code should be DRY. The whole section should be implemented in logic.js. Kindly work on that
-  From the instructions in pathwright handleClick is passed as a clickHandler property to ButtonPanel
- Then handleClick is passed as clickHandler props to each Button

- I haven't implemented props total or next to display as my logic needs to read a total and not independent values. therefore, is more efficient this way
- Not implemented the handle click function on the PanelButton as it needs the states and set states to finally render the information in screen, moving this function makes an inefficient management of the data and extends the numbers of functions to pass the data into App.js

**Review by enaburekhan — APPROVED** [review]
# Status: Approve:heavy_check_mark:
**Hi Sergio!
Congratulations:clap::clap:. Thank you for implementing the changes requested by the last reviewer.
Move on with the next project:walking:.**

Do have questions/comments? kindly drop them :point_down: and ping me on slack @ Eric Enaburekhan to respond.
Happy hacking:blush:
Eric Enaburekhan.

**Pull request merged**